tests_e2e/tests/extensions_disabled/extensions_disabled.py

```diff
@@ -47,27 +48,43 @@ def run(self):
         output = ssh_client.run_command("update-waagent-conf Extensions.Enabled=n", use_sudo=True)
         log.info("Disable completed:\n%s", output)
 
-        # From now on, extensions will time out; set the timeout to the minimum allowed(15 minutes)
+        # Set the timeout to the minimum allowed(15 minutes) so we can check that the extension fails before timeout
         log.info("Setting the extension timeout to 15 minutes")
         vm: VirtualMachineClient = VirtualMachineClient(self._context.vm)
-
         vm.update({"extensionsTimeBudget": "PT15M"})
 
         disabled_timestamp: datetime.datetime = datetime.datetime.utcnow() - datetime.timedelta(minutes=60)
 
         #
-        # Validate that the agent is not processing extensions by attempting to run CustomScript
+        # Validate that the agent is not processing extensions by attempting to run CSE & checking that provisioning fails fast
         #
-        log.info("Executing CustomScript; it should time out after 15 min or so.")
+        log.info("Executing CustomScript; the agent should report a VMExtensionProvisioningError without processing the extension")
         custom_script = VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.CustomScript, resource_name="CustomScript")
         try:
-            custom_script.enable(settings={'commandToExecute': "date"}, force_update=True, timeout=20 * 60)
-            fail("CustomScript should have timed out")
+            # Use mkdir in script settings so we can check if the command was executed
+            custom_script.enable(settings={'commandToExecute': "mkdir /var/lib/waagent/testdir"}, force_update=True, timeout=6 * 60)
+            fail("The agent should have reported an error processing the goal state")
         except Exception as error:
-            assert_that("VMExtensionProvisioningTimeout" in str(error)) \
-                .described_as(f"Expected a VMExtensionProvisioningTimeout: {error}") \
+            assert_that("VMExtensionProvisioningError" in str(error)) \
```

Comment:
is VMExtensionProvisioningError the type of the exception? if yes, we should handle this with "except VMExtensionProvisioningError" instead (instead of "in")

nice assertions, btw

Reply:
The exception type is: <class 'azure.core.exceptions.HttpResponseError'> from self._execute_async_operation